draft-barnes-hpke-pq.md

```diff
@@ -191,9 +192,42 @@ specified in {{ml-kem-iana-table}}.
 
 [[ TODO: Define HPKE API methods for the combination ]]
 
-# SHA-3 as an HPKE KDF
+# Single-Stage KDFs
 
-[[ TODO: Defer until draft-ietf-hpke-hpke has a suitable definition ]]
+This section defines HPKE KDFs for three eXtendable Output Functions (XOF) based
+on Keccak.  SHAKE is defined as part of the SHA-3 specification {{FIPS202}}, and
+the related TurboSHAKE and KangarooTwelve XOFs are a defined in
+{{!I-D.irtf-cfrg-kangarootwelve}}.
+
+The `Nh` values for the KDFs defined in this section are listed in
+{{kdfid-values}}.  The `Derive()` functions for each XOF are as follows, where
+`<SIZE>` is either 128 or 256:
+
+~~~ pseudocode
+def SHAKE<SIZE>.Derive(ikm, L):
+    return SHAKE<SIZE>(M = ikm, d = 8*L)
+
+def TurboSHAKE<SIZE>.Derive(ikm, L):
+    return TurboSHAKE<SIZE>(M = ikm, D = 0x1f, L)
+
+def KT<SIZE>.Derive(ikm, L):
+    return KT<SIZE>(M = ikm, C = "", L)
+~~~
+{: #kdf-derive title="Definition of single-stage KDFs" }
+
+| Value  | KDF           | Nh  | Two-Stage | Reference |
+|:-------|:--------------|-----|-----------|:----------|
+| 0x0000 | Reserved      | N/A | N/A       | RFC 9180  |
+| TBD    | SHAKE128      | 32  | N         | RFC XXXX  |
+| TBD    | SHAKE256      | 64  | N         | RFC XXXX  |
+| TBD    | TurboSHAKE128 | 32  | N         | RFC XXXX  |
+| TBD    | TurboSHAKE256 | 64  | N         | RFC XXXX  |
+| TBD    | KT128         | 32  | N         | RFC XXXX  |
```

Comment:
KT only makes sense for inputs longer than 24kB. Does HPKE ever pass such long inputs to the KDF?

Reply:
In principle, it's possible via the `info` parameter going into `KeySchedule`.  But it seems quite unlikely, and even then, there's a hard max of 64k, at least according to the current content of https://github.com/hpkewg/hpke/pull/16.  Given that, I would probably drop KT.